crates/db_connection_pool/src/postgrespool.rs

```diff
@@ -119,7 +119,6 @@ impl PostgresConnectionPool {
                     match pg_sslmode.to_lowercase().as_str() {
                         "disable" | "require" | "prefer" | "verify-ca" | "verify-full" => {
                             ssl_mode = pg_sslmode.as_str();
-                            connection_string.push_str(format!("sslmode={ssl_mode} ").as_str());
```

Comment:
it's not related to this line. But if connection_string has ssl_mode defined, that could be a double configuration for the ssl in the final connection_string?

Reply:
It was the case before this fix, but now the only place where sslmode added to the connection string is set is here

https://github.com/spiceai/spiceai/blob/02801593ff48c1b9af8a6380c065d143b63eed40/crates/db_connection_pool/src/postgrespool.rs#L151

Reply:
Updated the case with `pg_connection_string` to parse and extract sslmode and sslrootcert parameters, to process sll connection properly